client/src/components/Graph/D3Plot.vue

```diff
@@ -0,0 +1,307 @@
+        xScale = d3.scaleLinear();
+        yScale = d3.scaleLinear();
+      }
+      xScale = xScale
+        .domain([minXVal, maxXVal])
+        .range([leftPadding, w - rightPadding]);
+      yScale = yScale
+        .domain([minYVal, maxYVal])
+        .range([h - bottomPadding, bottomPadding]);
+
+      // Take each set of data points and put them on the plot
+      Object.values(datasets).forEach((dataset, i) => {
+        // Map each point to an array where it has the x, y value in each entry
+        const coordsArr = dataset.xs.map((val, index) => [
+          val,
+          dataset.ys[index],
+        ]);
+        svg.append('path')
+          .datum(coordsArr)
+          .attr('fill', 'none')
+          .attr('class', `line-${i}`)
+          .attr('stroke', colorGen(i))
+          .attr('stroke-width', 1.5)
+          .attr('d', d3.line()
+            .x((data) => xScale(data[0]))
+            .y((data) => yScale(data[1])));
+        svg.append('path');
+      });
+
+      const xAxis = d3.axisBottom(xScale);
+      const yAxis = d3.axisLeft(yScale);
+
+      svg.append('g')
+        .attr('id', 'x-axis')
+        .attr(
+          'transform',
+          `translate(0,${h - bottomPadding})`,
+        )
+        .call(xAxis);
+      svg.append('g')
+        .attr('id', 'y-axis')
+        .attr('transform', `translate(${leftPadding},0)`)
+        .call(yAxis);
+
+      // gridlines in x axis function
+      function make_x_gridlines() {
+        return d3.axisBottom(xScale)
+          .ticks(6);
+      }
+
+      // gridlines in y axis function
+      function make_y_gridlines() {
+        return d3.axisLeft(yScale)
+          .ticks(6);
+      }
+
+      // add the X gridlines
+      svg.append('g')
+        .attr('class', 'grid')
+        .attr('transform', `translate(0,${h - bottomPadding})`)
+        .call(make_x_gridlines()
+          .tickSize(-(h - bottomPadding * 2))
+          .tickFormat(''));
+
+      // add the Y gridlines
+      svg.append('g')
+        .attr('class', 'grid')
+        .attr('transform', `translate(${leftPadding},0)`)
+        .call(make_y_gridlines()
+          .tickSize(-(w - rightPadding - leftPadding))
+          .tickFormat(''));
+    },
+    /**
+     * Processes a latex string from the server into something that MathJax
+     * will happily accept 😁.
+     *
+     * @param {string} string the string to turn into a proper latex string
+     * which can be processed
+     * @returns {string} the output string to be provided to MathJax
+     */
+    processLatexString(rawLatex) {
+      return rawLatex
+        .trim()
+        .split(/\$/g)
+        .reduce((label, string, index) => {
+          if (index % 2 === 0) {
+            return `${label}\\textrm{${string}}`;
+          }
+          return label + string;
+        // Uses an empty string as the second argument to reduce so that it uses
+        // that as the starting point instead of the first array value.
+        }, '');
+    },
```

Comment:
We'll also probably need latex rendering in other places so this might make more sense in utils?

Reply:
Yes this could go in the utils folder. Maybe a file called `stringUtils`? Not sure where it would go in there or what it would be named.

Reply:
string-utils.js would work!

Reply:
This was done as requested.